<launch>
  <node name="position_control" pkg="youbot_position" type="position_control.py" />
  <node name="pose_forward_x" pkg="youbot_position" type="pose_node.py" />
  <node name="PID_X" pkg="pid" type="controller" >
    <!-- Topic for output of PID controller -->
    <param name="topic_from_controller" value="control_x" />
    <!-- Topic for state (x-component of pose) -->
    <param name="topic_from_plant" value="x_pid" />
    <!-- Topic for setpoint (desired x-component of pose) -->
    <param name="setpoint_topic" value="setpoint_x" />
    <param name="Kp" value="0.68" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.014" />
  </node>
  <node name="PID_Y" pkg="pid" type="controller" >
    <!-- Topic for output of PID controller -->
    <param name="topic_from_controller" value="control_y" />
    <!-- Topic for state (x-component of pose) -->
    <param name="topic_from_plant" value="y_pid" />
    <!-- Limit of error integral (set to default) -->
    <param name="windup_limit" value="1000"/>
    <!-- Topic for setpoint (desired x-component of pose) -->
    <param name="setpoint_topic" value="setpoint_y" />
    <param name="Kp" value="0.36" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.006" />
  </node>
</launch>
